Logstash PV sample test does not work on EKS

```
=== RUN   TestSamples/logstash-logstash_pv/Logstash_status_should_have_the_correct_status
Retries (15m0s timeout): ............................................................................................................................................................................................................................................................................................................
    step.go:51: 
        	Error Trace:	/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/utils.go:94
        	Error:      	Received unexpected error:
        	            	expected status {Version:8.12.1 ExpectedNodes:1 AvailableNodes:1 Health: ObservedGeneration:0 ElasticsearchAssociationsStatus: MonitoringAssociationStatus: Selector:} but got {Version:8.12.1 ExpectedNodes:1 AvailableNodes:0 Health:red ObservedGeneration:0 ElasticsearchAssociationsStatus: MonitoringAssociationStatus: Selector:common.k8s.elastic.co/type=logstash,logstash.k8s.elastic.co/name=d-g68c}
        	Test:       	TestSamples/logstash-logstash_pv/Logstash_status_should_have_the_correct_status
{"log.level":"error","@timestamp":"2024-02-11T22:49:11.995Z","message":"continuing with additional tests","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test Logstash status should have the correct status failed","error.stack_trace":"github.com/elastic/cloud-on-k8s/v2/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:52\ngithub.com/elastic/cloud-on-k8s/v2/test/e2e.TestSamples.func1\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/samples_test.go:40\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1595"}
{"log.level":"info","@timestamp":"2024-02-11T22:49:11.997Z","log.logger":"e2e","message":"Running eck-diagnostics","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","cluster":"eck-e2e-eks-eaof-452","test":"TestSamples/logstash-logstash_pv","step":"Logstash status should have the correct status"}
        --- FAIL: TestSamples/logstash-logstash_pv/Logstash_status_should_have_the_correct_status (900.00s)
```

The reason is that we attempt to create volumes with the default storage class which on EKS uses EBS volumes which are not available unless the EBS driver add-on is installed in the cluster which we don't do. 

I am not sure how this has ever worked. I have a suspicion that the logstash_pv test never ran, even on successful test runs: 

```
grep TestSamples/logstash  cloud-on-k8s-operator-nightly_build_445_go-eks.log | sed -n 's/.*Test":"\(TestSamples\/logstash-[^/"]*\)".*/\1/p' | sort | uniq
TestSamples/logstash-logstash
TestSamples/logstash-logstash_es
TestSamples/logstash-logstash_stackmonitor
TestSamples/logstash-logstash_svc
```

--- a/test/e2e/test/logstash/builder.go
+++ b/test/e2e/test/logstash/builder.go
@@ -153,17 +153,21 @@ func (b Builder) WithVolumeMounts(mounts ...corev1.VolumeMount) Builder {
 }
 
 func (b Builder) WithTestStorageClass() Builder {
+	hasDefaultVolumeClaim := false
 	for _, vc := range b.Logstash.Spec.VolumeClaimTemplates {
+		// a custom volume claim template is set patch it to use the test storage class
+		vc.Spec.StorageClassName = ptr.To(test.DefaultStorageClass)
 		if vc.Name == volume.LogstashDataVolumeName {
-			// a custom volume claim template is set we don't need to do anything
-			return b
+			hasDefaultVolumeClaim = true
 		}
 	}
 
-	// define the default volume claim with the e2e storage class
-	vc := volume.DefaultDataVolumeClaim.DeepCopy()
-	vc.Spec.StorageClassName = ptr.To[string](test.DefaultStorageClass)
-	b.Logstash.Spec.VolumeClaimTemplates = append(b.Logstash.Spec.VolumeClaimTemplates, *vc)
+	// define the default volume claim with the e2e storage class in case it is missing
+	if !hasDefaultVolumeClaim {
+		vc := volume.DefaultDataVolumeClaim.DeepCopy()
+		vc.Spec.StorageClassName = ptr.To[string](test.DefaultStorageClass)
+		b.Logstash.Spec.VolumeClaimTemplates = append(b.Logstash.Spec.VolumeClaimTemplates, *vc)
+	}
 	return b
 }
 

